### PR TITLE
New package: dlib-19.18

### PR DESCRIPTION
**Pull request by felix**

### CODE REVIEW

**Comment by Piraty** [comment]
hi,
- please squash your commits
- please use an order of variables that is consistent with other templates

**Comment by q66** [comment]
also protip: use rebase instead of merge when updating your branch

File: srcpkgs/dlib/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+homepage="http://dlib.net"
+distfiles="https://github.com/davisking/${pkgname}/archive/v${version}.tar.gz"
+checksum=e9a8540ad0ff84eda92dea7127cd7d078457caf7ced64400c25c9a0b8c96e242
+
+makedepends="$(vopt_if openblas openblas-devel) $(vopt_if x11 libX11-devel)"
+build_options="openblas x11"
+desc_option_openblas="Enable OpenBlas for performance"
+desc_option_x11="Enable X11 for GUI components"
+build_options_default="openblas x11"
+configure_args=Release
```
Comment (Piraty) [inline srcpkgs/dlib/template]: please move `configure_args` and `makedepends` (in that order) below `build_style`

**Comment by felix** [comment]
Thanks, I have not use void for a while so am a bit rusty. xlint does not seem to pick up the style errors mentioned above either.

**Comment by Piraty** [comment]
yes, we noticed xlint's fail as well.
@felix you got some other commit squashed as well, please cleanup

File: srcpkgs/dlib/template [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'dlib'
+pkgname=dlib
+version=19.18
+revision=1
+build_style=cmake
+configure_args=Release
```
Comment (Piraty) [inline srcpkgs/dlib/template:6]: that's not doing anything useful. [see here](https://github.com/davisking/dlib/blob/master/dlib/CMakeLists.txt#L48)

**Comment by Piraty** [comment]
I hope you don't mind i propose a modified version in #23007.

**Pull request closed without merging**